.github/actions/setup-zeebe/action.yml

```diff
@@ -1,113 +0,0 @@
-    required: false
-  secret_vault_roleId:
-    description: 'secret vault roleId'
-    required: false
-  secret_vault_secretId:
-    description: 'secret valut secret id'
-    required: false
-  docker:
-    description: If true, logs into Docker Hub using our CI account; helpful to prevent rate limiting
-    required: false
-    default: "true"
-  docker-token:
-    description: The token to use when logging into Docker Hub
-    required: false
-    default: "REGISTRY_HUB_DOCKER_COM_PSW_READ_ONLY"
-  docker-secret-path:
-    description: The path to Docker Hub secrets
-    required: false
-    default: "secret/data/products/zeebe/ci/zeebe"
-
-outputs: {}
-
-runs:
-  using: composite
-  steps:
-    - name: Secrets check
-      if: |
-        github.event.name == 'pull_request'
-        && github.event.pull_request.head.repo.full_name == 'camunda/camunda'
-        && (inputs.secret_vault_address == ''
-        || inputs.secret_vault_roleId == ''
-        || inputs.secret_vault_secretId == '')
-      shell: bash
-      run: |
-        echo "We will fail the action if it's an internal PR without the secrets. If this occurs it indicates a configuration failure that needs to be fixed."
-        exit 1
-    - name: Import Secrets
-      id: secrets
-      uses: hashicorp/vault-action@v3.0.0
-      if: |
-        inputs.secret_vault_address != ''
-        && inputs.secret_vault_roleId != ''
-        && inputs.secret_vault_secretId != ''
-      with:
-        url: ${{ inputs.secret_vault_address }}
-        method: approle
-        roleId: ${{ inputs.secret_vault_roleId }}
-        secretId: ${{ inputs.secret_vault_secretId }}
-        secrets: |
-          secret/data/products/zeebe/ci/zeebe ARTIFACTS_USR;
-          secret/data/products/zeebe/ci/zeebe ARTIFACTS_PSW;
-          ${{ inputs.docker-secret-path }} REGISTRY_HUB_DOCKER_COM_USR;
-          ${{ inputs.docker-secret-path }} ${{ inputs.docker-token }};
-    - uses: actions/setup-java@v4
-      if: inputs.java == 'true'
-      with:
-        distribution: 'temurin'
-        java-version: ${{ inputs.java-version }}
-    # Logging into Docker Hub will help prevent us being rate limited when using an anonymous account
-    - name: Login to Docker Hub
-      uses: docker/login-action@v3
-      if: |
-        inputs.docker == 'true'
-        && inputs.secret_vault_address != ''
-        && inputs.secret_vault_roleId != ''
-        && inputs.secret_vault_secretId != ''
-      with:
-        username: ${{ steps.secrets.outputs.REGISTRY_HUB_DOCKER_COM_USR }}
-        password: ${{ steps.secrets.outputs[inputs.docker-token] }}
-    # Use CI Nexus as co-located pull-through cache for Maven artifacts via ~/.m2/settings.xml
-    - name: 'Create settings.xml'
-      uses: s4u/maven-settings-action@v3.0.0
-      if: |
-        inputs.java == 'true'
-        && inputs.secret_vault_address != ''
-        && inputs.secret_vault_roleId != ''
-        && inputs.secret_vault_secretId != ''
```

Comment:
I think this and similar logic for other steps of the old `setup-zeebe` action got lost in the refactoring in this PR. This logic allows running CI on repositories forked from camunda/camunda that don't have access to Vault/secrets by falling back to Maven Central.

This feature is important for external contributors, see also https://github.com/camunda/camunda/issues/19222

This is also the reason why `setup-zeebe` handles secrets retrieval from Vault itself to be able to conditionally skip that and other related steps requiring secrets when detecting that run on a fork. Can you find a way to add this logic back to `setup-build` (implicitly also adding it to Operate/Tasklist etc)?

Reply:
Good catch!!
I can skip `camunda-nexus`config when running on forks, but I will probably have to introduce a new `setup-secrets` action to not break this logic. `setup-secrets` should not fail on forks and print a message that secrets cannot be retrieved. Not sure how to deal with expected (empty) outputs.

Reply:
As discussed, I have updated `steup-build` with same logic for fork PRs and will be used by default for all workflows. Team-dedicated accounts are used to preserve a similar quota distribution, but implemented to pave the way to a smooth transition to a unique account.

Reply:
For a reusable workflow, `github.workflows_ref` returns the name of the caller workflow. `camundacicamunda` thus needs read/write access to the `camunda/zeebe` repository as `zeebe-ci.yml` is called from `ci.yml` (zeebe DockerHub account will not be used in this case).

`preview-env-build-and-deploy.yml` results in deploying images to Harbor not DockerHub.

Reply:
> zeebe DockerHub account will not be used in this case
Makes sense to me 👍🏼